ls1intum/Artemis
# Enhancement/improve score accuracy

Fixes #2969 

### Checklist

### Motivation and Context
Artemis stores `result.score` as integer between 0 and (typically) 100 %.
In case `result.score` would have been a floating value (e.g. 2/3 points = 66.666666%), Artemis immediately rounded it to 67%, loosing precision when calculating the points later on using `result.score * exercise.maxPoints`. This led to rounding errors as described in #2969

### Description
Change `result.score` to double and only round on the client:
* Round scores (e.g. 67 %) in the client (user interface and various csv export) to 0 decimal places
* Round points (e.g. 10.5 points) in the client (user interface and various csv export) to 1 decimal place based on the max points of the exercise: `round(result.score * exercise.maxPoints, 1)`

### Steps for Testing
1. Check existing and new results to all kinds of exercises (course and exam)
2. Create manual (text, modeling, file-upload, programming) and automatic results (programming, quiz)
3. Make sure the score (%) and points calculation is correct: score in % with 0 decimal places (except specific places with 1 decimal place), points with 1 decimal place
4. Check the csv exports (for courses and exams) if they are correct
5. Test explicitly the case mentioned in the issue #2969

### Screenshots
Now we...

... correctly format the result string:
![grafik](https://user-images.githubusercontent.com/72132281/110260129-4513e900-7fab-11eb-9a29-ba92b353659e.png)

... properly format additional and inline feedback:
![grafik](https://user-images.githubusercontent.com/72132281/110260156-5a891300-7fab-11eb-8e00-d60485d4e41e.png)
![grafik](https://user-images.githubusercontent.com/72132281/110260160-5fe65d80-7fab-11eb-98de-dd72cacb3cb1.png)

... format the values for the course scores:
![grafik](https://user-images.githubusercontent.com/72132281/110260174-6d9be300-7fab-11eb-9250-dd56e36c1c2d.png)

... and export formatted values for course and exercise scores:
![grafik](https://user-images.githubusercontent.com/72132281/110260186-77bde180-7fab-11eb-8e7e-6f29ba14003b.png)
![grafik](https://user-images.githubusercontent.com/72132281/110260195-80aeb300-7fab-11eb-8a0c-5fa3833781ff.png)
